tools/src/main/java/org/neo4j/tools/applytx/DumpRecordsCommand.java

```diff
@@ -0,0 +1,192 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.tools.applytx;
+
+import io.airlift.airline.Arguments;
+import io.airlift.airline.Cli;
+import io.airlift.airline.Option;
+import io.airlift.airline.OptionType;
+
+import java.io.PrintStream;
+
+import org.neo4j.helpers.Provider;
+import org.neo4j.kernel.impl.core.Token;
+import org.neo4j.kernel.impl.store.RelationshipTypeTokenStore;
+import org.neo4j.kernel.impl.store.StoreAccess;
+import org.neo4j.kernel.impl.store.record.NodeRecord;
+import org.neo4j.kernel.impl.store.record.PropertyRecord;
+import org.neo4j.kernel.impl.store.record.Record;
+import org.neo4j.kernel.impl.store.record.RelationshipGroupRecord;
+import org.neo4j.kernel.impl.store.record.RelationshipRecord;
+import org.neo4j.tools.console.input.Command;
+import org.neo4j.tools.console.input.ConsoleInput;
+
+import static org.neo4j.tools.console.input.ConsoleUtil.airlineHelp;
+
+/**
+ * Able to dump records and record chains. Works as a {@link ConsoleInput} {@link Command}.
+ */
+public class DumpRecordsCommand implements Command
+{
+    private enum RecordType
+    {
+        node,
+        relationship;
+    }
+
+    private interface Action
+    {
+        void run( StoreAccess store, PrintStream out );
+    }
+
+    private final Cli<Action> cli;
+    private final Provider<StoreAccess> store;
+
+    @SuppressWarnings( "unchecked" )
+    public DumpRecordsCommand( String commandName, Provider<StoreAccess> store )
+    {
+        this.store = store;
+        this.cli = Cli.<Action>builder( commandName )
+                .withDescription( "Dump record contents" )
+                .withDefaultCommand( Help.class )
+                .withCommands( Help.class, DumpPropertyChain.class, DumpRelationshipChain.class,
+                        DumpRelationshipTypes.class )
+                .build();
+    }
+
+    @Override
+    public void run( String[] args, PrintStream out ) throws Exception
+    {
+        cli.parse( args ).run( store.instance(), out );
+    }
+
+    @Override
+    public String toString()
+    {
+        return airlineHelp( cli );
+    }
+
+    @io.airlift.airline.Command( name = "propchain", description = "Dump property chain information" )
+    public static class DumpPropertyChain implements Action
+    {
+        @Option( name = {"-t", "--type"}, type = OptionType.COMMAND, required = true, arity = 1 )
+        public String recordType;
```

Comment:
That's how I built it from the beginning, but it seems that Airline has a bug preventing multiple arguments to be specified consecutively like that :( I agree 100%. Maybe this could be solved using sub-groups or whatever it's called in airline. I'll give it a go.